src/stdlib/io.rs

```diff
@@ -69,4 +88,831 @@ pub fn load_io<'gc>(ctx: Context<'gc>) {
+                    .into_value(ctx)
+                    .into());
+            };
+
+            let whence_str = whence.map(|s| s.to_str().unwrap_or("cur")).unwrap_or("cur");
+            let offset = offset.unwrap_or(0);
+
+            let seek_from = match whence_str {
+                "set" => SeekFrom::Start(offset as u64),
+                "cur" => SeekFrom::Current(offset),
+                "end" => SeekFrom::End(offset),
+                _ => {
+                    return Err("bad argument #2 to 'seek' (invalid option)"
+                        .into_value(ctx)
+                        .into())
+                }
+            };
+
+            match file.inner() {
+                Either::Left(left) => {
+                    let mut left = left.borrow_mut();
+                    if let Some(ref mut left) = *left {
+                        match left.seek(seek_from) {
+                            Ok(position) => {
+                                file.set_read_position(position as usize);
+                                stack.replace(ctx, position as i64);
+                                Ok(CallbackReturn::Return)
+                            }
+                            Err(err) => {
+                                stack.replace(
+                                    ctx,
+                                    (Value::Nil, err.to_string(), err.raw_os_error().unwrap_or(0)),
+                                );
+                                Ok(CallbackReturn::Return)
+                            }
+                        }
+                    } else {
+                        Err("attempt to use a closed file".into_value(ctx).into())
+                    }
+                }
+                Either::Right(_) => {
+                    stack.replace(ctx, (Value::Nil, "Illegal seek", 29));
+                    Ok(CallbackReturn::Return)
+                }
+            }
+        }),
+    );
+
+    file.set_field(
+        ctx,
+        "write",
+        Callback::from_fn(&ctx, |ctx, _, mut stack| {
+            if stack.is_empty() {
+                return Err("bad argument #1 to 'write' (file expected)"
+                    .into_value(ctx)
+                    .into());
+            }
+
+            let Some(file) = stack.pop_front() else {
+                return Err("bad argument #1 to 'write' (file expected)"
+                    .into_value(ctx)
+                    .into());
+            };
+            let file = if let Value::UserData(file) = file {
+                if let Ok(file) = file.downcast_static::<IoFile>() {
+                    file
+                } else {
+                    return Err("bad argument #1 to 'write' (file expected)"
+                        .into_value(ctx)
+                        .into());
+                }
+            } else {
+                return Err("bad argument #1 to 'write' (file expected)"
+                    .into_value(ctx)
+                    .into());
+            };
```

Comment:
There should be a better way of managing the input arguments -- all of these errors have the exact same message, so it should be possible to shorten this?  (This also occurs in a number of other places.)